core/src/data/nullable.rs

```diff
@@ -0,0 +1,109 @@
+//! Submodule providing the `Nullable` enum and associated traits.
+
+use core::ops::Not;
+use std::ops::BitOr;
+
+#[derive(Debug, Clone, Copy, Hash)]
+/// Enum representing a nullable value.
+///
+/// # Implementations details
+/// This enum differs from the `Option` enum in that it does not represent
+/// a `Option::None` value, but rather a `Nullable::Null` SQL `NULL` value.
+///
+/// # Examples
+///
+/// ```
+/// use gluesql_core::data::Nullable;
+///
+/// let null: Nullable<i32> = Nullable::Null;
+///
+/// assert!(null.is_null());
+///
+/// let entry: Nullable<i32> = 10.into();
+///
+/// assert!(!entry.is_null());
+/// ```
+///
+pub enum Nullable<T> {
+    /// Represents an SQL `NULL` value.
+    Null,
+    /// Represents a non-`NULL` value.
+    Entry(T),
+}
+
+impl<T> Nullable<T> {
+    /// Returns `true` if the value is `Nullable::Null`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use gluesql_core::data::Nullable;
+    ///
+    /// let null: Nullable<i32> = Nullable::Null;
+    ///
+    /// assert!(null.is_null());
+    ///
+    /// let entry: Nullable<i32> = 10.into();
+    ///
+    /// assert!(!entry.is_null());
+    /// ```
+    pub fn is_null(&self) -> bool {
+        matches!(self, Nullable::Null)
+    }
```

Comment:
good. we now use `Nullable::is_null` only in unit tests. could we also remove this?

Reply:
You mean, adding a `#[cfg(test)]` on top of the method?

Reply:
oh, I saw some remaining is_null cases. I will close this conversation and look more on it.

currently I am focusing on..
<img width="632" alt="image" src="https://github.com/user-attachments/assets/a6aaf78c-2fa2-41e0-aa38-e60288e217d3">

to handle negate evaluation, I think that we can find more simple way to handle this but not yet I catched the point.

Reply:
This is a different is_null method, not the Nullable one. It may be reasonable to use the `Nullable` enum within these other structs in the future so as to uniform the handling of NULLs, but it is not the same method at this time.

Reply:
As of commit [a5e3c9b](https://github.com/gluesql/gluesql/pull/1577/commits/a5e3c9b68a06c4cb7c711b22682f5a60399cae48) `is_null` of the `Nullable` enum is a test-only method.